docs/server/multi-cluster.md

```diff
@@ -0,0 +1,392 @@
+
+When a workflow encounters divergence of workflow history, proper conflict resolution should be applied.
+
+In Multi-cluster Replication, workflow history events are modeled as a tree, as shown in the second example in [### Version History].
+
+Workflows which encounters divergence will have more than one history branches.
+Among all history branches, the history branch with the highest version is considered as the `current branch` and workflow mutable state is a summary of the current branch.
+Whenever there is a switch between workflow history branches, a complete rebuild of workflow mutable state will occur.
+
+## Zombie Workflows
+
+There is an existing contract that for any namespace & workflow ID combination, there can be at most one run (namespace & workflow ID & run ID) open / running.
+
+Multi-cluster Replication aims to keep the workflow state as up-to-date as possible among all participating Clusters.
+
+Due to the nature of Multi-cluster Replication, i.e. workflow history events are replicated asynchronously, different run (same namespace & workflow ID) can arrive at target Cluster at different times, sometimes out of order, as shown below:
+
+```
+| ------------- |          | ------------- |          | ------------- |
+|   Cluster A   |          | Network Layer |          |   Cluster B   |
+| ------------- |          | ------------- |          | ------------- |
+        |                          |                          |
+        | Run 1 Replication Events |                          |
+        | -----------------------> |                          |
+        |                          |                          |
+        | Run 2 Replication Events |                          |
+        | -----------------------> |                          |
+        |                          |                          |
+        |                          |                          |
+        |                          |                          |
+        |                          | Run 2 Replication Events |
+        |                          | -----------------------> |
+        |                          |                          |
+        |                          | Run 1 Replication Events |
+        |                          | -----------------------> |
+        |                          |                          |
+| ------------- |          | ------------- |          | ------------- |
+|   Cluster A   |          | Network Layer |          |   Cluster B   |
+| ------------- |          | ------------- |          | ------------- |
+```
+
+Since run 2 appears in Cluster B first, run 1 cannot be replicated as runnable due to rule `at most one run open` (see above), thus, `zombie` workflow state is introduced. Zombie state indicates a workflow which cannot be actively mutated by a Cluster (assuming corresponding namespace is active in this Cluster). A zombie workflow can only be changed by replication task.
```

Comment:
```suggestion
Since Run 2 appears in Cluster B first, Run 1 cannot be replicated as "runnable" due to the rule `at most one Run open` (see above), thus the "zombie" Workflow Execution state is introduced. 
A "zombie" state is one in which a Workflow Execution which cannot be actively mutated by a Cluster (assuming the corresponding Namespace is active in this Cluster). A zombie Workflow Execution can only be changed by a replication task.
```